src/overlay/Peer.cpp

```diff
@@ -917,7 +919,7 @@ Peer::recvMessage(StellarMessage const& stellarMsg)
         cat = "MISC";
     }
 
-    if (!mApp.getLedgerManager().isSynced() && stellarMsg.type() == TRANSACTION)
+    if (!mApp.getLedgerManager().isSynced() && ignoreIfOutOfSync)
```

Comment:
Looks like this will ignore transactions when we're out of sync (even when pull mode is disabled). Is that intended? It sounds like a good change but I was wondering since that'd change the behavior even when pull mode is disabled!

Reply:
this doesn't change the behavior, see previous condition `!mApp.getLedgerManager().isSynced() && stellarMsg.type() == TRANSACTION`

Reply:
Ah! Yes, that makes sense. I totally missed that.

Reply:
This looks good to me!